.github/workflows/ci.yml

```diff
@@ -0,0 +1,36 @@
+# This file is part of REANA.
+# Copyright (C) 2022 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details
+
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+    go-tests:
+        runs-on: ubuntu-20.04
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Setup Go
+              uses: actions/setup-go@v3
+              with:
+                  go-version: '>=1.18.X'
+            - run: go test ./... # TODO Replace with make test after #15
+
+    golangci:
+        runs-on: ubuntu-20.04
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Setup Go
+              uses: actions/setup-go@v3
+              with:
+                  go-version: '>=1.18.X'
+
+            - name: golangci-lint
```

Comment:
Very minor: would be good to stick to convention and to capitalise the step name